hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -115,6 +116,10 @@ std::recursive_mutex mdm_mutex;
 static volatile uint32_t gprs_timeout_start;
 static volatile uint32_t gprs_timeout_duration;
 
+// flags to check ucged command gives both rsrp and rsrq strings in its response
+static bool ucged_parse_rsrp_ok = false;
+static bool ucged_parse_rsrq_ok = false;
+
```

Comment:
These probably do not need to be in global space.  They can be calculated fresh each time _cbUCGED is called with local variables (stack memory), and the rsrp and rsrq can be invalidated.